doc/srv6/srv6_vpn.md

```diff
@@ -0,0 +1,284 @@
+# SRv6 VPN HLD
+
+# Overview
+
+Alibaba is deploying SRv6 VPN with SONiC based white box router at its edge of the network. The Alibaba's SRv6 VPN use case has been described in muliple conferences. The basic idea follows RFC 8986, which we use SRv6 as a network programming framework for applications.  VPN information is used as service anchor points for identifying service instances in the service plane. SRv6 policy could be used to provide paths to let applications reach service instances. The following diagram provides a summary on what Alibaba would like to achieve via SRv6 VPN.
+
+![image](images/1PGN.png)
+
+To help promoting SRv6's adopting in the industry, Alibaba team decides to upstream its deployed SRv6 VPN solution to SONiC community via 202305 release. We expect more contributors would join us to complete SRv6 VPN related features.
+
+# Design
+
+The SONiC SRv6 VPN deployable solution  requires changes from the following three communities.
+
+## FRR
+
+Besides recent FRR changes for SRv6 VPN contributed by Cisco team, Alibaba team needs to bring in the following FRR changes for a deployable solution. 
+
+1.  VRF routes leaking and performance tuning
+    
+2.  BGP advertise delay
+    
+3.  Conditional advertisement
+    
+4.  SRv6 Policy handling
+    
+
+FRR changes would be commited to FRR community.  SONiC 202305 release would pull in the corresponding  version of FRR release.
+
+The minimal FRR version for SRv6 VPN is 8.4, which you could get SRv6 VPN PoC functionalities. FRR version 8.5 or 9.0 would provide Alibaba deployment feature parity.
+
+## SAI
+
+Current SRv6 SAI doesn't have VPN support. SRv6 VPN SAI would be contributed by Cisco team. The SAI object model for SRv6 VPN is shown in the following diagram. The detail information about SRv6 VPN SAI APIs could be found in SAI's HLD document. ([https://github.com/rameshms-work/SAI/blob/rameshms/srv6\_vpn\_enhancements/doc/SAI-IPv6-Segment-Routing-VPN.md](https://github.com/rameshms-work/SAI/blob/rameshms/srv6_vpn_enhancements/doc/SAI-IPv6-Segment-Routing-VPN.md))
```

Comment:
This hyperlink to user specific work-space will become invalid when actual SAI work gets merged to community. Just mentioning here so as to track update to this hyperlink for long term validity in this HLD

Reply:
As I mentioned in the meeting, I will change it to SAI link which it gets merged.